xlb/operator/stepper/nse_stepper.py

```diff
@@ -91,92 +94,84 @@ def jax_implementation(self, f_0, f_1, bc_mask, missing_mask, timestep):
-            # NOTE: 'f_pre' is included here as an input to the BC functionals for consistency with the BC API,
-            # particularly when compared to post-collision boundary conditions (see below).
-
-            if _boundary_id == bc_struct.id_EquilibriumBC:
-                # Equilibrium boundary condition
-                f_post = self.EquilibriumBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_DoNothingBC:
-                # Do nothing boundary condition
-                f_post = self.DoNothingBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_HalfwayBounceBackBC:
-                # Half way boundary condition
-                f_post = self.HalfwayBounceBackBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_ZouHeBC_velocity:
-                # Zouhe boundary condition (bc type = velocity)
-                f_post = self.ZouHeBC_velocity.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_ZouHeBC_pressure:
-                # Zouhe boundary condition (bc type = pressure)
-                f_post = self.ZouHeBC_pressure.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_RegularizedBC_velocity:
-                # Regularized boundary condition (bc type = velocity)
-                f_post = self.RegularizedBC_velocity.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_RegularizedBC_pressure:
-                # Regularized boundary condition (bc type = velocity)
-                f_post = self.RegularizedBC_pressure.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_ExtrapolationOutflowBC:
-                # Regularized boundary condition (bc type = velocity)
-                f_post = self.ExtrapolationOutflowBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            elif _boundary_id == bc_struct.id_GradsApproximationBC:
-                # Reformulated Grads boundary condition
-                f_post = self.GradsApproximationBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
-            return f_post
+            f_result = f_post
+
+            if wp.static("EquilibriumBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["EquilibriumBC"]):
+                    f_result = self.EquilibriumBC.warp_functional(index, timestep, missing_mask, f_0, f_1, f_pre, f_post)
+
+            if wp.static("DoNothingBC" in self.active_bcs):
+                if _boundary_id == wp.static(boundary_condition_registry.bc_to_id["DoNothingBC"]):
```

Comment:
Please remove `boundary_condition_registry.` in all these if statements as bc_to_id is already defined above.